src/session/mod.rs

```diff
@@ -286,12 +285,19 @@ glib::wrapper! {
 }
 
 impl Session {
-    pub fn new(client_id: i32, database_info: DatabaseInfo) -> Self {
-        glib::Object::new(&[
+    pub fn new(client_id: i32, database_info: DatabaseInfo, me: tdgrand::types::User) -> Self {
+        let session = glib::Object::new::<Self>(&[
             ("client-id", &client_id),
             ("database-info", &BoxedDatabaseInfo(database_info)),
         ])
-        .expect("Failed to create Session")
+        .expect("Failed to create Session");
+
+        let me = User::from_td_object(me, &session);
+        session.user_list().insert_user(me.clone());
+        imp::Session::from_instance(&session).me.replace(Some(me));
```

Comment:
Why not passing the user in the Session's GObject constructor (glib::Object::new)?

Reply:
Because of a cyclic dependency.

Reply:
Oh, right.

Reply:
Yeah, this cycle is not easy to break. I tried it but gave up. :-P

Reply:
Well, then we could do two things: either passing the `me` user to UserList's constructor or inserting `me` in the UserList in the `ObjectImpl::set_property()` (I haven't tested either of those).

Reply:
I will test this tomorrow.

Reply:
Well, nevermind. I forgot that we still have to create the `User` in the constructor, so the cycle won't break either way. At this point I think your solution is the best one. I also checked if we can remove the `Session` from `User`'s constructor and we really can't. Resolving this.

Reply:
I was thinking, since tdlib guarantees that updates about types are always received before the relative id is returned to the app, I think that tdlib should also send an `Update::User` after sending the `GetMe` request. If this is the case then we may approach this in a different way: we can just save the id of the `me` user (via property) and then we can provide a method to quickly get the `me` user from the UserList.

Reply:
Just pushed a WIP commit demonstrating a way to do that.

Reply:
It seems fine, I'll do a proper review tomorrow.